File: CS12B/Lab2/FileReverse.java
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileWriter;
import java.io.IOException;
import java.io.PrintWriter;
import java.util.Scanner;

public class FileReverse {

	public static void main(String[] args) throws IOException {
		

		      int lineNumber = 0;

		      // check number of command line arguments is at least 2
		      if(args.length < 2){
		         System.out.println("Usage: FileCopy <input file> <output file>");
		         System.exit(1);
		      }

		      // open files
		      Scanner in = new Scanner(new File(args[0]));
		      PrintWriter out = new PrintWriter(new FileWriter(args[1]));

		      // read lines from in, extract and print tokens from each line
		      while( in.hasNextLine() ){
		         lineNumber++;

		         // trim leading and trailing spaces, then add one trailing space so 
		         // split works on blank lines
		         String line = in.nextLine().trim() + " "; 

		         // split line around white space 
		         String[] token = line.split("\\s+");  

		         // print out tokens       
		         int n = token.length;
		         for(int i=0; i<n; i++){
		            out.println(stringReverse(token[i]));
		         }
		      }

		      // close files
		      in.close();
		      out.close();
		      
		      
		      
		   }
		
	public static String stringReverse(String s){
		int l = s.length();
		String ret = "";
		for(int i=l-1;i>=0;i--){
			ret += s.charAt(i);
		}
		return ret;
	}

	}

File: CS12B/Lab3/FileReverse.c
/*FileReverse.c*/

#include <stdio.h>
#include <stdlib.h>
#include<string.h>

void stringReverse(char* s){
int i;
int j;
i=0;
j=strlen(s)-1;

while(j>i){
	char temp=s[j];
	s[j]=s[i];
	s[i]=temp;
	i++;
	j--;
}

}

int main(int argc, char* argv[]){
	FILE* in;  /* file handle for input */  
   FILE* out; /* file handle for output */
   char word[256]; /* char array to store words from input file */

   /* check command line for correct number of arguments */
   if( argc != 3 ){
      printf("Usage: %s <input file> <output file>\n", argv[0]);
      exit(EXIT_FAILURE);
   }

   /* open input file for reading */
   in = fopen(argv[1], "r");
   if( in==NULL ){
      printf("Unable to read from file %s\n", argv[1]);
      exit(EXIT_FAILURE);
   }

   /* open output file for writing */
   out = fopen(argv[2], "w");
   if( out==NULL ){
      printf("Unable to write to file %s\n", argv[2]);
      exit(EXIT_FAILURE);
   }

   /* read words from input file, print on separate lines to output file*/
   while( fscanf(in, " %s", word) != EOF ){
     	stringReverse(word);
	 fprintf(out, "%s\n", word);
   }

   /* close input and output files */
   fclose(in);
   fclose(out);

   return(EXIT_SUCCESS);

}
